src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DocumentHighlightHandler.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    [Shared]
+    [ExportLspMethod(Methods.TextDocumentDocumentHighlightName)]
+    internal class DocumentHighlightHandler : IRequestHandler<DocumentHighlightParams, DocumentHighlight[]>
+    {
+        private readonly LSPRequestInvoker _requestInvoker;
+        private readonly LSPDocumentManager _documentManager;
+        private readonly LSPProjectionProvider _projectionProvider;
+        private readonly LSPDocumentMappingProvider _documentMappingProvider;
+
+        [ImportingConstructor]
+        public DocumentHighlightHandler(
+            LSPRequestInvoker requestInvoker,
+            LSPDocumentManager documentManager,
+            LSPProjectionProvider projectionProvider,
+            LSPDocumentMappingProvider documentMappingProvider)
```

Comment:
Food for thought, we should probably merge mapping provider and projection provider since they both conceptually do the same thing (translate locations).

Reply:
https://github.com/dotnet/aspnetcore/issues/22288